src/pages/Main.jsx

```diff
@@ -36,9 +51,11 @@ const Main = () => {
           value={selectedLanguage}
           onChange={handleLanguageChange}
         >
-          <option value="ko">KR</option>
-          <option value="en">EN</option>
-          <option value="zh">CN</option>
+          {detectedLanguage === "ko" && (
+            <option value={detectedLanguage}>KO</option>
+          )}
+          {detectedLanguage === "en" && <option value="en">EN</option>}
+          {detectedLanguage === "zh" && <option value="zh">ZH</option>}
```

Comment:
줄일 수 있긴한데, 이 정도는 무방할 것 같습니다 :) 

만약에 사용하는 언어가 3개가 아니라 더 많아질경우, 
```suggestion
  
          {detectedLanguage && <option value={detectedLanguage}>detectedLanguage. toUpperCase()</option>}
     
```

한 줄로 줄일 수 있죠! 만약 detectedLanguage가 ""가 아니라 초기값이 있다면

```suggestion
  
          <option value={detectedLanguage}>detectedLanguage. toUpperCase()</option>
     
와 같이 줄일수도 있구요 :)

Reply:
일반화를 하는 방식의 코드 작성이 추후 유지 보수 및 확장성 고려에 용이합니다!